app/views/admin/problems/edit.html.erb

```diff
@@ -3,8 +3,18 @@
   <p>
     <%= f.error_messages %>
   </p>
+
   <%= render :partial => 'problem_fields', :locals => { :f => f } %>
   <p>
+    <p>
+      <%= f.label :runs_visible, "Видими ли са отговорите:"%>
+      <%= f.check_box :runs_visible, :disabled => true %>
+    </p>
     <%= f.submit "Обновяване" %><%= link_to "Отказ", admin_contest_problems_path(@problem.contest) %>
   </p>
-<% end -%>
\ No newline at end of file
+
+<% end -%>
+    <%if current_user.admin?%>
+      <%= button_to "Switch Visibility", {:action => "toggle_runs_visible"}, :method => "put"%>
+    <%end%>
```

Comment:
The indentation doesn't look right. Also some text explaining what is this button will be beneficial. Like

```html
<em>Да бъдат ли публични решенията на участниците?</em>
```

Also it is better to use the policy instead of checking if the user is an admin. See here for an example: https://github.com/valo/maycamp_arena/blob/master/app/views/admin/runs/_runs_table.html.erb#L25